src/Mod/Sketcher/Gui/ViewProviderSketch.h

```diff
@@ -779,6 +785,29 @@ class SketcherGuiExport ViewProviderSketch : public PartGui::ViewProvider2DObjec
 
     SoNodeSensor cameraSensor;
     int viewOrientationFactor; // stores if sketch viewed from front or back
+
+    /* - Grid things ---------------------------------------*/
+public:
+    bool getCameraCenter(const Gui::View3DInventorViewer* viewer);
+    bool getCameraMaxDimension(const Gui::View3DInventorViewer* viewer);
+    double getGridSize(const Gui::View3DInventorViewer* viewer);
+    SoSeparator* createGrid(bool cameraUpdate = false);
+    SoSeparator* createGridPart(double Step, int numberSubdiv, bool divLines, bool subDivLines, int pattern, SoBaseColor* color, int lineWidth = 1);
+
+    SbVec3f camCenterOnSketch;
+    float camMaxDimension;
+
+protected:
+    void Restore(Base::XMLReader& reader) override;
+    void handleChangedPropertyType(Base::XMLReader& reader, const char* TypeName, App::Property* prop) override;
+
+    SoSeparator* GridRoot;
+
+    bool gridSizeBlocker;

```

Comment:
What is the usage of this ?
I'm not sure it's needed as a member variable.
If needed, move to CoinManager.

Reply:
See 'onChanged' function.
It's to prevent cyclic call. As changing gridSize property trigger onChanged. Which triggers calculating gridSize.

Reply:
I think you can use a ` Base::ConnectionBlocker` as was done is `TaskSketcherGeneral`. ;)

Reply:
Got it.

Reply:
I don't understand how to use it here. What should I use as argument?
`TaskSketcherGeneral` use `    Base::ConnectionBlocker block(changedSketchView);`
Where
```
    changedSketchView = app->signalChangedObject.connect(boost::bind
        (&TaskSketcherGeneral::onChangedSketchView, this, bp::_1, bp::_2));
```
But here how do I get the signal value of onChanged ? It's not defined in vpSketch.

Reply:
Hmmm. Indeed I was too fast. I'm pretty sure there is a way but get it right from my memories. Will need some time to find out.

Reply:
@0penBrain 
when you talk about CoinManager do you mean EditModeCoinManager or a new class? In case of the former I would suggest to move all the grid handling to a helper class (e.g. SketchGridManager) and if needed add it as member to EditModeCoinManager.

Reply:
Calling @wwmayer. :) What is the correct way to temporary block a property change to trigger `onChanged` ? Is there some Blocker available. I thought it existed but can't find it. Thx

Reply:
> when you talk about CoinManager do you mean EditModeCoinManager or a new class?

New class. Sorry it was unclear.